File: DotNetGotchas/VBDN/SynchWithIntent/SynchronizeWithIntent/Bacteria.vb
Imports System.Threading

'Bacteria.vb
Public Class Bacteria

    Private Shared bacteriaCount As Integer

    Private Shared Sub IncreaseCount()
        Console.WriteLine( _
         "IncreaseCount called by {0} at {1}", _
         AppDomain.GetCurrentThreadId(), _
         DateTime.Now.ToLongTimeString())

        bacteriaCount += 1

        Thread.Sleep(2000)
        ' Used for illustration purpose
    End Sub

    Public Sub New()
        SyncLock GetType(Bacteria)
            IncreaseCount()
        End SyncLock
    End Sub

End Class

File: DotNetGotchas/CSharp/Synchronizing/Modified1/SynchOnType/Bacteria.cs
//Bacteria.cs
using System;

namespace SynchOnType
{
	public class Bacteria
	{
		private static int bacteriaCount;

		private static void IncreaseCount()
		{
			Console.WriteLine(
				"IncreaseCount called by {0} at {1}",
				AppDomain.GetCurrentThreadId(),
				DateTime.Now.ToLongTimeString());

			bacteriaCount++;

			System.Threading.Thread.Sleep(2000);
			// Used for illustration purpose
		}

		public Bacteria()
		{
			lock(GetType())
			{
				IncreaseCount();
			}
		}
	}
}
